src/common/syncjournaldb.cpp

```diff
@@ -978,37 +978,42 @@ bool SyncJournalDb::deleteFileRecord(const QString &filename, bool recursively)
 }
 
 
-SyncJournalFileRecord SyncJournalDb::getFileRecord(const QString &filename)
+bool SyncJournalDb::getFileRecord(const QString &filename, SyncJournalFileRecord *rec)
```

Comment:
I'm not a fan of this kind of API.
Maybe do something like std::optional<SyncJournalFileRecord> instead

Reply:
This is how the call site looks with this API:
```
SyncJournalFileRecord rec;
if (!journal->getFileRecord("path", &rec)
    emit error();
else if (rec.isValid())
    qDebug() << rec._path;
```

This is how it would like if we passed the success as the out argument (takes one more line):
```
bool ok = false;
SyncJournalFileRecord rec = journal->getFileRecord("path", &ok);
if (!ok)
    emit error();
else if (rec.isValid())
    qDebug() << rec._path;
```

And this is how it could be with some custom return data structure (what a more specific implementation of std::optional could look like):
```
JournalResult<SyncJournalFileRecord> result = journal->getFileRecord("path");
if (!result.isSuccess)
    emit error();
else if (result.value.isValid())
    qDebug() << result.value._path;
```
But this would ideally also mean I have to rewrite most of the DB access code for things to be consistent. So that's why I chose to go with bool return value for now. Let me know if you think otherwise.

Reply:
Or we could start using exceptions 😜